### ISSUE TITLE
Crash and freeze caused upon `:wq` with large file 

### ISSUE THREAD

**Issue by rgasper** [helix-editor/helix#4453]
### Summary

I have a large file, an about 130K line csv. After making some changes with helix in many lines in the line 300 -> line 650 region, when I save using `:wq`, helix crashes and stops writing to disk around line 12750, effectively deleting the remaining 120,000 lines in the file. After the crash, helix is stuck controlling the terminal, catching all inputs; the screen displays whatever the helix view was when `:wq` was run. The terminal session has to be killed by the emulator or OS. 

If it helps debugging, the lines in the file average just about 100 bytes, but are a bit shorter in the beginning few thousand lines, averaging closer to 70 bytes per line.

As noted in the title, the curious part of this crash is that if I run `:w` then `:q` after, there are no problems.

Running helix in very very verbose mode, and then editing the file, then I see no messages of relevance in the log. Just this right at the same time as the crash:
```
2022-10-24T14:39:21.520 helix_term::job [DEBUG] waiting on jobs...
``` 

### Reproduction Steps

Change many lines in a large file, then `:wq`. Helix should crash and hang on the terminal, blocking keyboard inputs.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2022-10-24T14:39:21.520 helix_term::job [DEBUG] waiting on jobs...
```

</details>

### Platform

macOS Monterey 12.5

### Terminal Emulator

iterm2

### Helix Version

helix 22.08 (ee94031f)

**Comment by the-mikedavis** [comment]
Duplicate of https://github.com/helix-editor/helix/issues/3975

This was fixed in https://github.com/helix-editor/helix/pull/2267
